### DISCUSSION TITLE
Chart component recommendation (Office hours: Sept 24, 2020)

### DISCUSSION THREAD

**Discussion by sang-yeo** [twilio-labs/paste#781] [Office hours] [answered]
The specific questions I have are...

- How does Paste components interact with a chart in a setting / configuration? 
- Chart component recommendation

**Background:** we are planning to test a concept with customers on customizable alerts. Our research tells us that customers like to see a previous data in a chart when they setup an alert so that they can easily compare and understand.  

[Lo-fi mock](https://drive.google.com/file/d/1CaDJYZBYFKBX3wEV8Qo1lTO2Pb5J530u/view?usp=sharing)

**Comment by corinne-nagel** [comment, marked as answer]
Hey @sang-yeo , we can chat about these tomorrow but some thoughts:

What is the purpose a chart in a create/setup page? Is this the appropriate stage to show this?

Charts should use the existing library used in Console (Highcharts). We'll have data visualization color tokens and guidance by the end of the year to plug in, but if you want to look at JT's current work on Messaging Insights and match colors to hers for now that would be your best bet.

**Reply by sang-yeo** [reply]
Yep, looking forward to discussing with you all. The main purpose of showing a dynamic chart is we are trying to deliver intuitive experience to our customers when they set up an alert rule. For example, when a customer visits the Alert page to set up an alert for weekly data usage limit, the customer likes to see the current pattern so that they can set up an alert based on that data. You can find similar patterns in Google Cloud and AWS Cloud Watch. 

I talked to JT about the chart and received her work.

**Reply by sang-yeo** [reply]
Hey @corinne-nagel do you have any update on the **data visualization color tokens and guidance** you mentioned above?

**Reply by serifluous** [reply]
Hey @sang-yeo, the data visualization tokens got pushed back on our roadmap after some re-prioritization, unfortunately. We'll be kicking them off in our sprint that starts today. I'll make sure we follow up with you here when we add the guidelines.

**Reply by sang-yeo** [reply]
hi @serifluous following up with you on the data visualization tokens (aka, charts). Any updates?

**Reply by serifluous** [reply]
Hey @sang-yeo , data visualization tokens are actually going to come in through a contribution from the V&V team now. You can follow along in the slack channel #tmp-pattern-charting.

Do you still need them for the customizable alerts feature you shared above?

**Reply by sang-yeo** [reply]
Thanks Sarah! Yes, there will be an area for the graph in the Alarms setup page. 
Do you happen to know when the contribution will go live?

**Reply by kate-porter** [reply]
Hey @sang-yeo! We don't have a date for when that will go live yet, as the V&V team are just getting started on it and will also need to balance that work with upcoming Signal commitments. Definitely feel free to join the #tmp-pattern-charting channel in Slack to follow along and share any work you've done, too.
